### DISCUSSION TITLE
trace with zipkin

### DISCUSSION THREAD

**Discussion by fanjianfeng2014** [apache/apisix#6310] [Q&A]
when I enable zipkin plugin, the service name always is default ，not the specific upstream service name.
![image](https://user-images.githubusercontent.com/7496672/153805673-e15d6314-9497-4481-9ab8-e13860623968.png)
the other question :
We also use zipkin in dapr,  the headers filled by the W3C standard,but it diffierent to APISIX，how to compatibale?
https://www.w3.org/TR/trace-context/

**Comment by dmsolr** [comment]
first, the service name is irrelated to the upstream service name, so I cannot get your means.
then, do the headers mean the zipkin `b3-propagation` headers?
If so, I think they are compatible. Because the APISIX fully follows the [Zipkin B3 specification](https://github.com/openzipkin/b3-propagation)

**Reply by fanjianfeng2014** [reply]
first, the service name is irrelated to the upstream service name, so I cannot get your means.
Is the zipkin report "request -proxy - rewrite" 3 sequential ? not like gateway - serviceA - serviceB ,only in APISIX self?

**Reply by dmsolr** [reply]
That looks like you don't enable Zipkin on your upstream service. This picture shows the APISIX enabled Zipkin only. 
For example, we deployed like Client -> APISIX -> Server, can see following:
![image](https://user-images.githubusercontent.com/29735230/153825444-9d03f2de-c5f4-4cc8-bd75-f211dc17b3eb.png)
